### PR TITLE
Logout functionality

### PR DESCRIPTION
**Pull request by xlasserre**
Logout functionality. 
https://trello.com/c/zqBlhFLJ/20-3-as-a-user-i-should-be-able-to-logout-from-the-application

The logout is on the same page as the profile page, so I added some layout for the profile edition, but nothing of its logic.

### CODE REVIEW

File: src/components/ProfileForm/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      component={CustomInput}
+    />
+    <Field
+      name="password"
+      inputLabel="Password"
+      inputType="password"
+      component={CustomInput}
+    />
+    <CustomButton
+      title="SAVE CHANGES" color={black}
```
Comment (bernabe9) [inline src/components/ProfileForm/index.js]: `color` prop should be in a new line.

When all the props don't fit in one line you should use one line per prop

File: src/containers/ProfileScreen/index.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import { connect } from 'react-redux';
+import { func } from 'prop-types';
+
+import ProfileForm from '../../components/ProfileForm';
+import styles from './styles';
+import { logout, editProfile } from '../../actions/userActions';
+
+class ProfileScreen extends React.Component {
+  constructor(props) {
+    super(props);
```
Comment (bernabe9) [inline src/containers/ProfileScreen/index.js]: `props` are not necessary

File: src/containers/TargetPointsScreen/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            </Text>
+          </TouchableOpacity>
+        </View>
+      </View>
+    );
+  }
+}
+
+MainScreen.propTypes = {
+  navigator: object
```
Comment (bernabe9) [inline src/containers/TargetPointsScreen/index.js]: this props seems to be required

File: src/containers/TargetPointsScreen/index.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import { object } from 'prop-types';
+
+import navigatorStyles from '../../components/common/Navigator/styles';
+import styles from './styles';
+
+class MainScreen extends React.Component {
+  static navigatorStyle = navigatorStyles;
+
+  constructor(props) {
+    super(props);
```
Comment (bernabe9) [inline src/containers/TargetPointsScreen/index.js]: props are not necessary

File: src/containers/TargetPointsScreen/index.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
+  }
+
+  onNavigatorEvent(event) {
+    if (event.type == 'NavBarButtonPress') {
```
Comment (bernabe9) [inline src/containers/TargetPointsScreen/index.js]: Please use the Identity Operator (===) instead

File: src/containers/ProfileScreen/index.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            </TouchableOpacity>
+          </View>
+        </View>
+      </View>
+    );
+  }
+}
+
+ProfileScreen.propTypes = {
+  userLogout: func
```
Comment (bernabe9) [inline src/containers/ProfileScreen/index.js]: isnt required?

File: src/containers/ProfileScreen/index.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import {
+  View, TouchableOpacity,
```
Comment (bernabe9) [inline src/containers/ProfileScreen/index.js]: Use one line per module

**Review by bernabe9 — CHANGES_REQUESTED** [review]
Looks good!

I added some code style comments

File: src/components/ProfileForm/index.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+      onPress={handleSubmit(editProfile)} disabled={submitting}
+      customStyle={styles.changeButton}
+    />
+    { error && <Text>{error}</Text>}
+  </View>
+);
+
+ProfileForm.propTypes = {
+  handleSubmit: PropTypes.func.isRequired,
```
Comment (agch-dev) [inline src/components/ProfileForm/index.js]: Remember to import `PropTypes` on their own like `{ bool, func, string }`

File: src/containers/ProfileScreen/styles.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  screen: {
+    alignItems: 'stretch',
+    flex: 1,
+    flexDirection: 'column',
```
Comment (agch-dev) [inline src/containers/ProfileScreen/styles.js]: Is this necessary? `'column'` is the default value.

File: src/containers/ProfileScreen/styles.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+  logOutButton: {
+    marginTop: 20
+  },
+
+  logOutTxt: {
+    alignSelf: 'center',
+    fontWeight: 'bold'
+  }
+
```
Comment (agch-dev) [inline src/containers/ProfileScreen/styles.js]: Remove extra line

File: src/components/ProfileForm/index.js [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+import CustomInput from '../common/CustomInput';
+import CustomButton from '../common/CustomButton';
+import { validations, editProfile } from '../../utils/constraints';
+import { black } from '../../constants/styleConstants';
+import styles from './styles';
+
+const ProfileForm = ({
+  handleSubmit, error, submitting, editProfile
+}) => (
+  <View>
```
Comment (agch-dev) [inline src/components/ProfileForm/index.js:15]: Have you checked if a keyboardAwareView is necessary here?

**Review by agch-dev — CHANGES_REQUESTED** [review]
Great Work!

File: src/components/ProfileForm/index.js (hunk 11 above)
Comment (xlasserre) [reply src/components/ProfileForm/index.js:15]: I haven't still but because it's not part of the logout functionality.

**Review by agch-dev — APPROVED** [review]

**Review by bernabe9 — APPROVED** [review]

**Pull request merged**